itest/lnd_wallet_import_test.go

```diff
@@ -550,6 +556,35 @@ func runWalletImportAccountScenario(ht *lntest.HarnessTest,
 	}
 	dave.RPC.ImportAccount(importReq)
 
+	// Try to import an account with the same name but with a different
+	// key scope. It should return an error.
+	otherAddrType := walletrpc.AddressType_TAPROOT_PUBKEY
+	if addrType == walletrpc.AddressType_TAPROOT_PUBKEY {
+		otherAddrType--
+	}
+
+	listReq = &walletrpc.ListAccountsRequest{
+		Name:        "default",
+		AddressType: otherAddrType,
+	}
+	listResp = carol.RPC.ListAccounts(listReq)
+	require.Len(ht, listResp.Accounts, 1)
+
+	carolAccountOtherAddrType := listResp.Accounts[0]
+
+	errAccountExists := fmt.Sprintf(
+		"account '%s' already exists", importedAccount,
+	)
+
+	importReq = &walletrpc.ImportAccountRequest{
+		Name:              importedAccount,
+		ExtendedPublicKey: carolAccountOtherAddrType.ExtendedPublicKey,
+		AddressType:       otherAddrType,
+	}
+	err := dave.RPC.ImportAccountAssertErr(importReq)
+	require.Contains(ht, err.Error(), errAccountExists,
```

Comment:
nit: Could use `require.ErrorContains` to avoid the `err.Error()`. Also, I think the `require` library creates enough context in its error messages that the additional string isn't necessary and everything can fit on one line.